CHANGELOG updates for 3.4.0

krobelus opened the pull request:
I used the command from #8092 to list issues/PRs with missing changelog entries,
and went through most of them.

Each issue gets three lines
- subject
- URL
- verdict

If the verdict ends with ", ignoring", I added it the the ignored list in the changelog.

The issues are grouped by verdict, with the interesting/leftover ones
on top (obviously no need to double check *everything*).

The "gh" script is already a quantum leap we can still find better
ways to share the changelog burden.
I noticed that there are many minor updates that can probably be
ignored.  Filtering them out doesn't take much time but it adds up,
especially if it's a single person doing it.

---

Issue/PR title: Make --no-config mode more comfortable
https://github.com/fish-shell/fish-shell/issues/8493
in-flight

Issue/PR title: Debian fish binary package difficult to install
https://github.com/fish-shell/fish-shell/issues/7845
TODO

Issue/PR title: Work around `setpgid` error on older Apple platforms
https://github.com/fish-shell/fish-shell/issues/8153
this extends 7474, which wasn't listed either, we should probably mention it?

Issue/PR title: Abbr -q return status inconsistent
https://github.com/fish-shell/fish-shell/issues/8431
fixes return status of "abbr -q", should we mention this?

Issue/PR title: math: (n n): incorrect error
https://github.com/fish-shell/fish-shell/issues/8511
improved error output, which is very nice but too minor?

Issue/PR title: Fish autocomplete error on iOS procursus
https://github.com/fish-shell/fish-shell/issues/8205
niche fix, ignoring

Issue/PR title: Fix `fish_key_reader` wrapper check
https://github.com/fish-shell/fish-shell/issues/8271
minor update to not create a harmless alias for fish_key_reader, ignoring

Issue/PR title: funced dosn't like backslash escapes in function names
https://github.com/fish-shell/fish-shell/issues/8289
minor escaping fix, ignoring

Issue/PR title: Hide whatis database building from the user
https://github.com/fish-shell/fish-shell/issues/8310
not something many users would notice in the first place, ignoring

Issue/PR title: Duplicated "Type 'help argparse' for related documentation" for argparse
https://github.com/fish-shell/fish-shell/issues/8368
minor update to error message, ignoring

Issue/PR title: Variable highlight color does not span lines
https://github.com/fish-shell/fish-shell/issues/8444
very obscure fix, ignoring

---

Issue/PR title: Add --function to `read`
https://github.com/fish-shell/fish-shell/issues/8295
added to existing entry (565)

Issue/PR title: Added completions for ethtool
https://github.com/fish-shell/fish-shell/issues/8283
added to existing entry

Issue/PR title: Add dart completion
https://github.com/fish-shell/fish-shell/issues/8315
added to existing entry

Issue/PR title: Add common lisp completions(sbcl/roswell)
https://github.com/fish-shell/fish-shell/issues/8330
added to existing entry

Issue/PR title: Fix st issue with shift+tab
https://github.com/fish-shell/fish-shell/issues/8354
added to existing entry (8352)

Issue/PR title: Support vi-mode cursors in Foot Terminal 
https://github.com/fish-shell/fish-shell/issues/8391
added to existing entry (8167)

Issue/PR title: Completions pager should redraw if the subbed completion wraps/unwraps the line
https://github.com/fish-shell/fish-shell/issues/8405
added to existing entry (8509)

---

Issue/PR title: Binding escape as user binding breaks escape sequence bindings (arrows, etc)
https://github.com/fish-shell/fish-shell/issues/8428
added new entry

Issue/PR title: Windows "color" command completion
https://github.com/fish-shell/fish-shell/issues/8483
added new entry

Issue/PR title: Doesn't build when using netbsd curses on Linux
https://github.com/fish-shell/fish-shell/issues/8087
added new entry

Issue/PR title: Don't override linker
https://github.com/fish-shell/fish-shell/issues/8152
added new entry

Issue/PR title: Add completions for `git-sizer`
https://github.com/fish-shell/fish-shell/issues/8156
added new entry

Issue/PR title: `d3ceba107e88b6c6e1a0358ebcb30366aeef653f` causes issues with repainting multi-line prompt
https://github.com/fish-shell/fish-shell/issues/8163
added new entry

Issue/PR title: Builtin math ncr can be extremely slow
https://github.com/fish-shell/fish-shell/issues/8170
added new entry

Issue/PR title: Completion sometimes missing the last token
https://github.com/fish-shell/fish-shell/issues/8175
added new entry

Issue/PR title: `set -S` should mark read-only variables
https://github.com/fish-shell/fish-shell/issues/8179
added new entry

Issue/PR title: Errors when trying to autocomplete (invalid) UTF-8 escapes
https://github.com/fish-shell/fish-shell/issues/8195
added new entry

Issue/PR title: Always use LC_NUMERIC=C internally
https://github.com/fish-shell/fish-shell/issues/8204
added new entry

Issue/PR title: Slow interaction between backgrounding, universal variables, and repainting
https://github.com/fish-shell/fish-shell/issues/8209
added new entry

Issue/PR title: Unsetting `$fish_emoji_width` doesn't clear the cached width
https://github.com/fish-shell/fish-shell/issues/8274
added new entry

Issue/PR title: If prompt ends in an empty line, the commandline is inserted at the width of the line before
https://github.com/fish-shell/fish-shell/issues/8298
added new entry

Issue/PR title: assertion normal_exited() failed related to paged builtin help
https://github.com/fish-shell/fish-shell/issues/8308
added new entry

Issue/PR title: colors don't kick in for ls on macOS Big Sur, Monterey (and maybe FreeBSD)
https://github.com/fish-shell/fish-shell/issues/8309
added new entry

Issue/PR title: Adds sub-command clear-session to history command. Issue #5791
https://github.com/fish-shell/fish-shell/issues/8337
added new entry (as 5791)

Issue/PR title: Display local branches before unique remote branches in git completion
https://github.com/fish-shell/fish-shell/issues/8338
added new entry

Issue/PR title: Fix delete-key in st
https://github.com/fish-shell/fish-shell/issues/8352
added new entry

Issue/PR title: sigsegv on set --show variable (when LANG is set to fr_FR.utf8)
https://github.com/fish-shell/fish-shell/issues/8358
added new entry

Issue/PR title: Add clasp completion
https://github.com/fish-shell/fish-shell/issues/8373
added new entry

Issue/PR title: `cargo run --example` completions break with nested example directories
https://github.com/fish-shell/fish-shell/issues/8429
added new entry

Issue/PR title: argparse completions
https://github.com/fish-shell/fish-shell/issues/8434
added new entry

Issue/PR title: Stop linking to StackOverflow
https://github.com/fish-shell/fish-shell/issues/8495
added new entry

Issue/PR title: fish_key_reader ^C warning isn't right
https://github.com/fish-shell/fish-shell/issues/8510
added new entry

Issue/PR title: Use `--almost-all` in `la` function
https://github.com/fish-shell/fish-shell/issues/8519
added new entry

---

Issue/PR title: improve the experience of using fish over mosh
https://github.com/fish-shell/fish-shell/issues/1363
listed as 8376, ignoring

Issue/PR title: incomplete man page completions
https://github.com/fish-shell/fish-shell/issues/8305
listed as 8309, ignoring

Issue/PR title: Support "$(cmd)" command substitution without line splitting
https://github.com/fish-shell/fish-shell/issues/8059
listed as 159, ignoring

Issue/PR title: fish_config: Read colorschemes from .theme files
https://github.com/fish-shell/fish-shell/issues/8127
listed as 8132, ignoring

Issue/PR title: funced: edit the whole file, not just the function definition
https://github.com/fish-shell/fish-shell/issues/8130
listed as 391, ignoring

Issue/PR title: builtin cd: print error about broken symlink
https://github.com/fish-shell/fish-shell/issues/8270
listed as 8264, ignoring

Issue/PR title: fix man completion for BSD's mandoc
https://github.com/fish-shell/fish-shell/issues/8306
listed as 8305, ignoring

Issue/PR title: Don't escape tildes that come from custom completions
https://github.com/fish-shell/fish-shell/issues/8441
listed as 8441, ignoring

Issue/PR title: Use `cargo run --example` to get list of examples
https://github.com/fish-shell/fish-shell/issues/8446
listed as 8429, ignoring

---

Issue/PR title: Node completion: add v8 sparkplug option
https://github.com/fish-shell/fish-shell/issues/8118
update to existing completions, ignoring

Issue/PR title: Add zypper subcommands completion
https://github.com/fish-shell/fish-shell/issues/8183
update to existing completions, ignoring

Issue/PR title: completion nmap: suppress warning when local scripts folder exists
https://github.com/fish-shell/fish-shell/issues/8184
update to existing completions, ignoring

Issue/PR title: add missing `git commit` completions
https://github.com/fish-shell/fish-shell/issues/8191
update to existing completions, ignoring

Issue/PR title: Updated ping completions
https://github.com/fish-shell/fish-shell/issues/8192
update to existing completions, ignoring

Issue/PR title: Add `--function` to `set` completion
https://github.com/fish-shell/fish-shell/issues/8202
update to existing completions, ignoring

Issue/PR title: completion: support `--no` prefixes for mpv flag options
https://github.com/fish-shell/fish-shell/issues/8219
update to existing completions, ignoring

Issue/PR title: complete "mpc load"
https://github.com/fish-shell/fish-shell/issues/8241
update to existing completions, ignoring

Issue/PR title: Add and fix completions for new options
https://github.com/fish-shell/fish-shell/issues/8243
update to existing completions, ignoring

Issue/PR title: Fix completions/ls.fish
https://github.com/fish-shell/fish-shell/issues/8249
update to existing completions, ignoring

Issue/PR title: Fix completions/coredumpctl.fish and add new complete
https://github.com/fish-shell/fish-shell/issues/8256
update to existing completions, ignoring

Issue/PR title: completions/git: Handle "1 .T" & "1 AT" files
https://github.com/fish-shell/fish-shell/issues/8311
update to existing completions, ignoring

Issue/PR title: completions/xbps-query: add missing `-p` completions
https://github.com/fish-shell/fish-shell/issues/8323
update to existing completions, ignoring

Issue/PR title: Update ldapsearch.fish
https://github.com/fish-shell/fish-shell/issues/8326
update to existing completions, ignoring

Issue/PR title: small fix completions/duply.fish
https://github.com/fish-shell/fish-shell/issues/8327
update to existing completions, ignoring

Issue/PR title: Update ip.fish
https://github.com/fish-shell/fish-shell/issues/8334
update to existing completions, ignoring

Issue/PR title: Fix ant completion
https://github.com/fish-shell/fish-shell/issues/8344
update to existing completions, ignoring

Issue/PR title: Update dmesg completions
https://github.com/fish-shell/fish-shell/issues/8365
update to existing completions, ignoring

Issue/PR title: No hints for -g|--global and -U|--universal flags for abbr command
https://github.com/fish-shell/fish-shell/issues/8367
update to existing completions, ignoring

Issue/PR title: Updated systemd-analyze completions
https://github.com/fish-shell/fish-shell/issues/8381
update to existing completions, ignoring

Issue/PR title: vmctl completion function call needs to be quoted
https://github.com/fish-shell/fish-shell/issues/8406
update to existing completions, ignoring

Issue/PR title: pabcnetcclear command completion update
https://github.com/fish-shell/fish-shell/issues/8480
update to existing completions, ignoring

---

Issue/PR title: document `--no-config`
https://github.com/fish-shell/fish-shell/issues/8176
doc update, ignoring

Issue/PR title: Theme demo needs to be adjusted so that only unmatched quote is an error
https://github.com/fish-shell/fish-shell/issues/8260
doc update, ignoring

Issue/PR title: no error about wrong >>? redirection operator
https://github.com/fish-shell/fish-shell/issues/8380
doc update, ignoring

Issue/PR title: set -l works outside of command block
https://github.com/fish-shell/fish-shell/issues/8385
doc update, ignoring

Issue/PR title: Some enhancements to "for" and "while" loop pages
https://github.com/fish-shell/fish-shell/issues/8409
doc update, ignoring

Issue/PR title: Html docs: Remove link underlines again?
https://github.com/fish-shell/fish-shell/issues/8439
doc update, ignoring

Issue/PR title: Old-style options support "=" assignment operator in complete builtin
https://github.com/fish-shell/fish-shell/issues/8457
doc update, ignoring

Issue/PR title: Document prompt_hostname
https://github.com/fish-shell/fish-shell/issues/8522
doc update, ignoring

---

Issue/PR title: edit_command_buffer: use "command" to ignore any functions with the same name
https://github.com/fish-shell/fish-shell/issues/8221
only helps broken systems, ignoring

Issue/PR title: Prepend command to cat
https://github.com/fish-shell/fish-shell/issues/8287
only helps broken systems, ignoring

Issue/PR title: Make less version check compatible with older Fish
https://github.com/fish-shell/fish-shell/issues/8299
only helps broken systems, ignoring

Issue/PR title: Skip leading `command` in `__fish_man_page`
https://github.com/fish-shell/fish-shell/issues/8447
only helps broken systems, ignoring

Issue/PR title: fish_config doesn't work without curses module
https://github.com/fish-shell/fish-shell/issues/8487
only helps broken systems, ignoring

---

Issue/PR title: fix 'socket file name too long' error
https://github.com/fish-shell/fish-shell/issues/8128
test fix with long tempdirs (macOS), not really user-visible, ignoring

Issue/PR title: Give tests a more generic name
https://github.com/fish-shell/fish-shell/issues/8449
not user-visible, ignoring

Issue/PR title: string tests sometimes failing on macOS (Github Actions)
https://github.com/fish-shell/fish-shell/issues/8353
not user-visible, ignoring

Issue/PR title: history merge test fails on OpenBSD
https://github.com/fish-shell/fish-shell/issues/6477
not user-visible, ignoring

---

Issue/PR title: Obtain Deno completions from itself
https://github.com/fish-shell/fish-shell/issues/8471
update to an unreleased feature (7138), ignoring

Issue/PR title: `string length --visible` performance
https://github.com/fish-shell/fish-shell/issues/8253
update to an unreleased feature, ignoring

Issue/PR title: Backspace character is ignored when calculating string widths
https://github.com/fish-shell/fish-shell/issues/8277
update to an unreleased feature, ignoring

Issue/PR title: `fish_config choose` leaves previous right prompt in place
https://github.com/fish-shell/fish-shell/issues/8314
update to an unreleased feature, ignoring

Issue/PR title: parenthesis characters outer of $(command substitution) in string cause error
https://github.com/fish-shell/fish-shell/issues/8394
update to an unreleased feature, ignoring

Issue/PR title: Parser bug with command substitutions in strings inside parenthesis
https://github.com/fish-shell/fish-shell/issues/8500
update to an unreleased feature, ignoring

Issue/PR title: fish_config: silently doesn't set color schemes.
https://github.com/fish-shell/fish-shell/issues/8419
regression, not in any release, ignoring

Issue/PR title: :program: in sphinx doesn't link
https://github.com/fish-shell/fish-shell/issues/8438
regression, not in any release, ignoring

Issue/PR title: __fish_seen_argument.fish throws exception when autocompleting
https://github.com/fish-shell/fish-shell/issues/8478
regression, not in any release, ignoring

---

Issue/PR title: Fix typo in abbr docs
https://github.com/fish-shell/fish-shell/issues/8280
typofix, ignoring

Issue/PR title: Fix typo in `set_colors` command documentation
https://github.com/fish-shell/fish-shell/issues/8321
typofix, ignoring

Issue/PR title: Typo funcions -> functions
https://github.com/fish-shell/fish-shell/issues/8257
typofix, ignoring

---

Issue/PR title: remove make_pair
https://github.com/fish-shell/fish-shell/issues/8206
no behavior change, ignoring

Issue/PR title: replace push_back with emplate_back
https://github.com/fish-shell/fish-shell/issues/8222
no behavior change, ignoring

Issue/PR title: clang-tidy: remove pointless virtual
https://github.com/fish-shell/fish-shell/issues/8224
no behavior change, ignoring

Issue/PR title: change value to rvalue reference
https://github.com/fish-shell/fish-shell/issues/8227
no behavior change, ignoring

Issue/PR title: convert const ref to rvalue ref
https://github.com/fish-shell/fish-shell/issues/8228
no behavior change, ignoring

Issue/PR title: clang-tidy: use for range loops
https://github.com/fish-shell/fish-shell/issues/8229
no behavior change, ignoring

Issue/PR title: fix deleted constructors
https://github.com/fish-shell/fish-shell/issues/8230
nno behavior change, ignoring

Issue/PR title: clang-tidy: const reference conversions
https://github.com/fish-shell/fish-shell/issues/8231
no behavior change, ignoring

Issue/PR title: clang-tidy: simplify two bool returns
https://github.com/fish-shell/fish-shell/issues/8235
no behavior change, ignoring

Issue/PR title: clang-tidy: replace size comparisons with empty
https://github.com/fish-shell/fish-shell/issues/8237
no behavior change, ignoring

Issue/PR title: clang-tidy: replace NULL with nullptr
https://github.com/fish-shell/fish-shell/issues/8239
no behavior change, ignoring

Issue/PR title: add constexpr
https://github.com/fish-shell/fish-shell/issues/8252
no behavior change, ignoring

Issue/PR title: __fish_seen_subcommand_from and __fish_seen_argument update
https://github.com/fish-shell/fish-shell/issues/8430
no behavior change (apart from a regression that's fixed), ignoring

Issue/PR title: Run fish_indent on all non-test .fish files
https://github.com/fish-shell/fish-shell/issues/8476
no behavior change, ignoring

Issue/PR title: Use test command instead of bracket command
https://github.com/fish-shell/fish-shell/issues/8477
no behavior change, ignoring

Issue/PR title: Fix code scanning alert - Wrong type of arguments to formatting function
https://github.com/fish-shell/fish-shell/issues/8521
no behavior change, ignoring

Issue/PR title: clang-tidy: replace push_back with emplace_back
https://github.com/fish-shell/fish-shell/issues/8236
no behavior change, ignoring

---

faho commented:
> this extends 7474, which wasn't listed either, we should probably mention it?

This one is moot because we've decided to not support these macOS versions anyway - see 83799228fe656020b9b7d35c87003275533550c2. This is because it will no longer build because we're using fstatat now - 71a0d839a7d6426650d096412b1a06e86353577f.

We've also decided to drop workarounds immediately, so it's pretty permanent and not just a matter of reverting the fstatat commit - see 01919f1785844ef973cf70b24328951fbb318891 and parents.

So it's superseded by the line about dropping support for macOS < 10.10.

zanchey commented on CHANGELOG.rst:
```diff
@@ -96,13 +105,23 @@ Interactive improvements
 - Custom completions are now also loaded for commands that contain tildes or variables like ``~/bin/fish`` or ``$PWD/fish`` (:issue:`8442`).
-- Command lines spanning multiple lines will not be overwritten by the completion pager when it fills the entire terminal (:issue:`8509`).
+- Command lines spanning multiple lines will not be overwritten by the completion pager when it fills the entire terminal (:issue:`8509`, :issue:`8405`).
+- When redrawing a multiline prompt, the old prompt was not fully cleared. This has been corrected (:issue:`8163`).
```
How about "the old prompt is now properly cleared (8163)"

zanchey commented on CHANGELOG.rst:
```diff
@@ -96,13 +105,23 @@ Interactive improvements
+- Command lines spanning multiple lines will not be overwritten by the completion pager when it fills the entire terminal (:issue:`8509`, :issue:`8405`).
+- When redrawing a multiline prompt, the old prompt was not fully cleared. This has been corrected (:issue:`8163`).
+- Interactive completion would occasionally ignore the last word on the command line due to a race condition. This has been fixed (:issue:`8175`).
+- fish would print errors when trying to autoload completions for command names with embedded null bytes, because those names are truncated by C library routines. This has been fixed by refusing to autoload commands with null bytes (:issue:`8195`).
```
I think this can be dropped, it's a pretty pathological case.

zanchey commented on CHANGELOG.rst:
```diff
@@ -96,13 +105,23 @@ Interactive improvements
+- When redrawing a multiline prompt, the old prompt was not fully cleared. This has been corrected (:issue:`8163`).
+- Interactive completion would occasionally ignore the last word on the command line due to a race condition. This has been fixed (:issue:`8175`).
+- fish would print errors when trying to autoload completions for command names with embedded null bytes, because those names are truncated by C library routines. This has been fixed by refusing to autoload commands with null bytes (:issue:`8195`).
+- When a short-lived fish process writes a universal variable, propagation was slow in some cases. This has been sped up (:issue:`8209`).
```
How about "Propagation of universal variables from a fish process that is closing is faster"?

zanchey commented on CHANGELOG.rst:
```diff
@@ -96,13 +105,23 @@ Interactive improvements
+- Interactive completion would occasionally ignore the last word on the command line due to a race condition. This has been fixed (:issue:`8175`).
+- fish would print errors when trying to autoload completions for command names with embedded null bytes, because those names are truncated by C library routines. This has been fixed by refusing to autoload commands with null bytes (:issue:`8195`).
+- When a short-lived fish process writes a universal variable, propagation was slow in some cases. This has been sped up (:issue:`8209`).
+- The commandline buffer is drawn in the correct place if the prompt has a trailing empty line (:issue:`8298`).
```
I've tried to stick with "command line" to refer to the current command line buffer, but there's a few other places this needs fixing too.

krobelus replied:
done, I'll add a commit fixing the other occurrences in 3.4.0

zanchey commented on CHANGELOG.rst:
```diff
@@ -96,13 +105,23 @@ Interactive improvements
+- fish would print errors when trying to autoload completions for command names with embedded null bytes, because those names are truncated by C library routines. This has been fixed by refusing to autoload commands with null bytes (:issue:`8195`).
+- When a short-lived fish process writes a universal variable, propagation was slow in some cases. This has been sped up (:issue:`8209`).
+- The commandline buffer is drawn in the correct place if the prompt has a trailing empty line (:issue:`8298`).
+- When the pager invoked by a builtin's ``--help`` flag was killed by a signal, fish would crash with an assertion failure, which has been fixed (:issue:`8308`).
```
This is pretty niche! I think it can go, or trimmed down to "A crash when reading help text has been fixed"

zanchey commented on CHANGELOG.rst:
```diff
@@ -96,13 +105,23 @@ Interactive improvements
 
 New or improved bindings
 ^^^^^^^^^^^^^^^^^^^^^^^^
+- Binding :kbd:`Escape` used to break arrow key bindings, because their key codes are prefixed by the escape character. This has been fixed (:issue:`8428`).
```
What about "Escape can now be bound without breaking arrow-key bindings"?

zanchey reviewed (commented):
Wow! This is amazing work, thank you! I agree almost completely with your choices on what to ignore. I think 8447 could probably go in, though I'm happy to add that at some stage.

I've made some suggestions below to align the entries a bit more to the style I prefer. I've never explicitly codified it, but there's a few terminology choices and I try to stick to present tense, positive statements (avoiding "no longer crashes" or similar), and user observable changes (like "X can now Y", rather than "Made X work with Y"). I'm happy to make these edits after merge if you would prefer.

faho commented on CHANGELOG.rst:
```diff
@@ -78,15 +79,23 @@ Scripting improvements
 - Tilde characters (``~``) produced by custom completions are no longer escaped when applied to the command line, making it easier to use the output of a recursive ``complete -C`` in completion scripts (:issue:`4570`).
+- ``set --show`` now marks read-only variables (:issue:`8179`).
+- ``LANG=fr_FR.utf8 set --show`` used to crash due to an invalid format string in a translation string, which has been fixed (:issue:`8358`).
+- Builtins ``printf`` and ``math`` used to read the environment variable ``LC_NUMERIC`` to use locale-specific number format. They have been changed to always use ``LC_NUMERIC=C``. In particular ``math``'s decimal separator is now always ".", never "," (:issue:`8204`).
```
This is wrong - `math` *always* used `.`. Using `,` doesn't work because that's the argument separator - `ncr(5,2)` is n choose r with n = 5 and r = 2. Even if we worked around that (Excel uses `;` as the argument separator in German!), it would still break expressions brought onto a system with a different locale - this is a common problem with badly ported linux games and shaders! So the only tenable thing to do for any script thing that reads arithmetic expressions is to always use `.` as the decimal separator.

So that is what `math` did from the very beginning. It did that by doing `setlocale(LC_NUMERIC, "C")` while math was being run.

Instead, we now do `setlocale(LC_NUMERIC, "C")` *always*, so for fish internally numbers are always "." and then we override it in the one place where we want localized numbers, which is printf. (tho tbh I would rather we don't do localized numbers there either, but that's a different change)

This makes `math` faster by quite a bit and, on systems supporting `uselocale`, improves thread safety while not affecting printf performance a lot. It also makes it easier for fish to stay correct - 99% of the time we want unlocalized numbers with `.`.

-------

If anything, the user-visible effect is that math is faster.

An additional effect of this was accidentally fixing detection of when we can use posix_spawn on glibc systems with a comma-using locale, which was later properly fixed with https://github.com/fish-shell/fish-shell/commit/f0d4fd85b116a30319c1b42f1f79c33ebaa69441. This increases performance of launching external processes on those systems by ~50% (as in you can run 1.5x the processes in the same time, with things like `command echo`).

krobelus replied:
Wow, thanks for the explanation. I'll drop it since math performance isn't so important. The posix_spawn detection fix is nice though!

krobelus commented:
Thanks for the review! I think I have addressed all comments, so this should be ready. (I haven't resolved threads since I don't know how to do that with the github API)

The ignore list is really essential, we should definitely keep sharing this information in future (in whatever form is convenient).

(Of course we might have relevant changes that don't have issues, or issues that we forget to milestone but that shouldn't happen too often.)

> I think 8447 could probably go in

Yeah definitely, I confused this with one of the changes that prepend "command" somewhere to protect against false friends..
I removed it from the ignore list, now there are 4 issues left ;)

krobelus commented on CHANGELOG.rst:
```diff
@@ -151,16 +176,21 @@ Improved terminal support
 ----------------
 - The minimum version of CMake required to build fish was raised to 3.5.0.
 - The CMake installation supports absolute paths for ``CMAKE_INSTALL_DATADIR`` (:issue:`8150`).
+- A build failure when using NetBSD curses on non-NetBSD platforms has been fixed (:issue:`8087`).
```
yep, much better

krobelus commented on CHANGELOG.rst:
```diff
@@ -151,16 +176,21 @@ Improved terminal support
 - The minimum version of CMake required to build fish was raised to 3.5.0.
 - The CMake installation supports absolute paths for ``CMAKE_INSTALL_DATADIR`` (:issue:`8150`).
+- A build failure when using NetBSD curses on non-NetBSD platforms has been fixed (:issue:`8087`).
+- The build system no longer tries to override the default linker with LLVM's ``lld`` or GNU ``gold`` (:issue:`8152`).
```
the "different" sounds a bit lonely, but it seems fine

krobelus commented on CHANGELOG.rst:
```diff
@@ -151,16 +176,21 @@ Improved terminal support
 - fish's escape sequence removal now also knows Tmux's wrapped escapes.
 - Vi cursor shaping and $PWD reporting is now also enabled on foot (:issue:`8422`).
+- Add a workaround for the ``Terminal.app`` terminal: export ``CLICOLOR`` to ``ls`` because new BSD ``ls`` does not enable colors otherwise (:issue:`8309`).
+- Add workarounds for the ``st`` terminal to make the :kbd:`Delete` and :kbd:`Shift-Tab` keys work (:issue:`8352`, :issue:`8354`).
```
sounds good

krobelus commented on CHANGELOG.rst:
```diff
@@ -132,16 +153,20 @@ Completions
 - Add missing completions for the ``-p`` option of ``xbps-query``.
 - The ``fish_is_nth_token`` function, which is particularly useful in completions for identifying the token number within the command line, replaces various internal functions to do the same (:issue:`8008`).
+- When using BSD ``mandoc``, man-page completions would fail to offer man pages that have aliases like ``infocmp(1, 1M, 1m)``. This has been fixed (:issue:`8305`).
+- ``cargo run --example`` completions now offer examples from nested directories (:issue:`8429`).
```
yeah, definitely

krobelus commented on CHANGELOG.rst:
```diff
@@ -132,16 +153,20 @@ Completions
+- ``git`` completions display local branches before unique remote branches (:issue:`8338`).
 - Add missing completions for the ``-p`` option of ``xbps-query``.
 - The ``fish_is_nth_token`` function, which is particularly useful in completions for identifying the token number within the command line, replaces various internal functions to do the same (:issue:`8008`).
+- When using BSD ``mandoc``, man-page completions would fail to offer man pages that have aliases like ``infocmp(1, 1M, 1m)``. This has been fixed (:issue:`8305`).
```
Oh this is not about autogenerated completions but about completions for `man` itself.
I rephrased it but it can definitely be dropped.

zanchey replied:
Oh, right, that can be dropped then.

krobelus commented on CHANGELOG.rst:
```diff
@@ -132,16 +153,20 @@ Completions
   - ``zef`` (:issue:`8114`)
 
 - Improvements to many completions, especially for ``git`` aliases (:issue:`8129`) and subcommands (:issue:`8134`).
+- ``git`` completions display local branches before unique remote branches (:issue:`8338`).
```
yeah, dropped

krobelus commented on CHANGELOG.rst:
```diff
@@ -78,15 +79,23 @@ Scripting improvements
 - ``set --query`` can now query whether a variable is a path variable via ``--path`` or ``--unpath`` (:issue:`8494`).
 - Tilde characters (``~``) produced by custom completions are no longer escaped when applied to the command line, making it easier to use the output of a recursive ``complete -C`` in completion scripts (:issue:`4570`).
+- ``set --show`` now marks read-only variables (:issue:`8179`).
+- ``LANG=fr_FR.utf8 set --show`` used to crash due to an invalid format string in a translation string, which has been fixed (:issue:`8358`).
```
yeah, so crash fixes are usually not that interesting

krobelus commented on CHANGELOG.rst:
```diff
@@ -78,15 +79,23 @@ Scripting improvements
 - On MacOS terminals that are not granted permissions to access a folder, ``cd`` would print a spurious "rotten symlink" error, which has been corrected to "permission denied" (:issue:`8264`).
 - Since fish 3.0, for-loops would trigger a variable handler an additional time before the loop was entered. This has been corrected (:issue:`8384`).
 - ``math`` now correctly prints negative values and values larger than ``2**31`` when in hex or octal bases (:issue:`8417`).
+- ``math ncr`` has been sped up (:issue:`8170`).
```
Right, I didn't realize that so I was wondering where the speedup is

zanchey commented:
Thanks, that's excellent.
